src/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -752,9 +755,12 @@ private final static boolean isVCFStream(final InputStream stream, final String
         return new LazyGenotypesContext.LazyData(genotypes, header.getSampleNamesInOrder(), header.getSampleNameToOffset());
     }
 
-    private final String[] INT_DECODE_ARRAY = new String[10000];
+    private final int LARGE_SIZE = 10000;
+    private final String[] INT_DECODE_ARRAY = new String[2*LARGE_SIZE];
     private final int[] decodeInts(final String string) {
         final int nValues = ParsingUtils.split(string, INT_DECODE_ARRAY, ',');
+        if ( nValues > LARGE_SIZE )
+            log.warn("Decoding " + nValues + " integers in " + name);
```

Comment:
I think this warning needs to be more clear, eg add something like "This warning is associated with decoding genotype annotations that have a very large number of values, typically due to the presence of a very large number of alternate alleles." so anyone who encounters it in the future has some idea of what's happening.